internal/eval/store_eval.go

```diff
@@ -316,3 +316,264 @@ func evalSETEX(args []string, store *dstore.Store) *EvalResponse {
 
 	return evalSET(newArgs, store)
 }
+
+// GETEX key [EX seconds | PX milliseconds | EXAT unix-time-seconds |
+// PXAT unix-time-milliseconds | PERSIST]
+// Get the value of key and optionally set its expiration.
+// GETEX is similar to GET, but is a write command with additional options.
+// The GETEX command supports a set of options that modify its behavior:
+// EX seconds -- Set the specified expire time, in seconds.
+// PX milliseconds -- Set the specified expire time, in milliseconds.
+// EXAT timestamp-seconds -- Set the specified Unix time at which the key will expire, in seconds.
+// PXAT timestamp-milliseconds -- Set the specified Unix time at which the key will expire, in milliseconds.
+// PERSIST -- Remove the time to live associated with the key.
+// The RESP value of the key is encoded and then returned
+// evalGET returns response.RespNIL if key is expired or it does not exist
+func evalGETEX(args []string, store *dstore.Store) *EvalResponse {
+	if len(args) < 1 {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongArgumentCount("GETEX"),
+		}
+	}
+
+	var key = args[0]
+
+	var exDurationMs int64 = -1
+	var state = Uninitialized
+	var persist = false
+	for i := 1; i < len(args); i++ {
+		arg := strings.ToUpper(args[i])
+		switch arg {
+		case Ex, Px:
+			if state != Uninitialized {
+				return &EvalResponse{
+					Result: nil,
+					Error:  diceerrors.ErrSyntax,
+				}
+			}
+			i++
+			if i == len(args) {
+				return &EvalResponse{
+					Result: nil,
+					Error:  diceerrors.ErrSyntax,
+				}
+			}
+
+			// If an option is provided instead of expire time return a syntax error
+			if opt := strings.ToUpper(args[i]); opt == Persist || opt == Px || opt == Ex {
```

Comment:
This case seems a bit weird. Since we are already in the case for Ex, Px still we are checking for the same opt and invalidating it. Can you elaborate more?

Reply:
`opt` actually refers to the argument after the current one which is basically the expiry time. If it's not a valid number the error returned is `diceerrors.ErrIntegerOutOfRange` but in case we have something like: `GETEX key EX PERSIST 4` it's a Syntax Error so I'm manually checking if the it's one of those commands to know whether it should be a syntax error or not.

Reply:
But now that you've mentioned it, I checked how redis implements it and it isn't a syntax error so maybe I should change that. Should I also add unit tests for that case?
![Screenshot_20241014_153744](https://github.com/user-attachments/assets/ba0f6ab5-a5d3-4a2c-bf99-dddba4db82d2)

Reply:
Hey @Ehijoe thanks for the response, got the logic. 
yes, you can make it redis compliance and add a unit test as well. Thanks a lot.

Reply:
Okay @AshwinKul28, I've made the behaviour similar to redis. Should be good to go now right?